inc/project.class.php

```diff
@@ -1256,6 +1295,19 @@ function prepareInputForAdd($input) {
 
 
    function prepareInputForUpdate($input) {
+      if (isset($input['auto_percent_done']) && $input['auto_percent_done']) {
+         unset($input['percent_done']);
+      }
+      if (isset($input['projects_id']) && $input['projects_id'] > 0) {
+         if (self::checkCircularRelation($input['id'], $input['projects_id'])) {
+            Session::addMessageAfterRedirect(__('Circular relation found. Parent not updated.'), false,
+                                          ERROR);
+            unset($input['projects_id']);
+         }
+      }
+      if (isset($input['projects_id']) && ($input['projects_id'] != $this->fields['projects_id'])) {
```

Comment:
A PHP notice is triggered when defining a parent on a project that had no parent.
```suggestion
      if ($this->fields['projects_id'] > 0 && isset($input['projects_id'])
          && ($input['projects_id'] != $this->fields['projects_id'])) {
```
```
  *** PHP Notice(8): Undefined index: auto_percent_done
  Backtrace :
  inc/project.class.php:1968                         
  inc/project.class.php:252                          Project::recalculatePercentDone()
  inc/commondbtm.class.php:1433                      Project->post_updateItem()
  front/project.form.php:94                          CommonDBTM->update()
```